packages/headless/src/controllers/insight/folded-result-list/headless-insight-folded-result-list.ts

```diff
@@ -0,0 +1,52 @@
+import {insightFoldedResultAnalyticsClient} from '../../../features/folding/folding-insight-analytics-actions';
+import {loadCollection} from '../../../features/folding/insight-folding-actions';
+import {InsightEngine} from '../../../insight.index';
+import {
+  buildCoreFoldedResultList,
+  FoldingOptions,
+  FoldedCollection,
+  FoldedResult,
+  FoldedResultList,
+  FoldedResultListOptions,
+  FoldedResultListState,
+} from '../../core/folded-result-list/headless-core-folded-result-list';
+
+export type {
+  FoldedResultListOptions,
+  FoldedResultListState,
+  FoldedResultList,
+  FoldingOptions,
+  FoldedCollection,
+  FoldedResult,
+};
+
+export interface FoldedResultListProps {
```

Comment:
You should rename this something like `InsightFoldedResultListProps` to avoid confusion.